test/Exploratory_Benchmarks/polyglot-sources/exploratory-benchmark-java-helpers/src/main/java/org/enso/exploratory_benchmark_helpers/MapHelpers.java

```diff
@@ -103,7 +102,7 @@ public static Storage<?> mapCallback(
     int n = storage.size();
     Builder builder =
         expectedType == null
-            ? new InferredBuilder(n, problemAggregator)
+            ? Builder.getForType(null, n, problemAggregator)
```

Comment:
While I appreciate `getForType` returning inferred builder if the type is null (unknown), for places like this I wonder if it won't be more readable to have a separate `Builder.getInferred(...)` static method. IMHO it will make the code more readable (it is more clear vs a slightly 'magical' `null` argument).


I'd still keep the `null` argument for cases where we get some expected type and it is either known or unknown - I think there it works OK. But here, IMHO a more explicit approach would be more readable.

Although if you think it's bad to have 2 ways to construct the inferred builder then I'm also OK to keep it as is

Reply:
No don't disagree will add it and make the explicit nulls call it.